### PR TITLE
Fix: UI of Saving Account Details when offline

### LINKED ISSUE #461: In Saving Account Details, when offline, error message is not aligned properly
**Summary:**

If a User is offline, and user goes to Saving Account Details, the error is not aligned well

**Steps to reproduce:**

Home --> Accounts --> Any Saving Accounts

**Expected behavior:**

Error message should be aligned properly.

**Observed behavior:**

Error message is showing at top center position, icon and text are colliding. 

**Device and Android version:**

Coolpad Cool 1 (Android 6.0.1 Marshmallow)

**Screenshots:**

screenshot_20171015-220345

@naman14 @therajanmaurya @dilpreet96 Can I work on this issue ?

### PR DESCRIPTION
**Pull request by sashank27**
Fixes #461 

Please Add Screenshots If any UI changes.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

**Comment by sashank27** [comment]
@dilpreet96 @therajanmaurya @tarun0 please review

**Comment by dilpreet2028** [comment]
@sashank27 , we need to display the view in centre of the screen.

**Comment by sashank27** [comment]
Fixed the display. @dilpreet96 please review

screenshot_20180206-233520

File: app/src/main/res/layout/fragment_saving_account_details.xml [hunk 1]
```diff
@@ -408,24 +405,24 @@
...
-                    android:focusable="true"
                     android:id="@+id/tv_help_line_number"
-                    android:layout_height="wrap_content"
                     android:layout_width="wrap_content"
+                    android:layout_height="wrap_content"
+                    android:clickable="true"
+                    android:focusable="true"
                     android:text="@string/help_line_number"
-                    android:textColor="@color/blue"/>
+                    android:textColor="@color/blue" />
```
Comment (luckyman20) [inline app/src/main/res/layout/fragment_saving_account_details.xml]: @sashank27 Why you have added this space?

File: app/src/main/res/layout/fragment_saving_account_details.xml [hunk 2]
```diff
@@ -184,220 +181,220 @@
...
-                                android:textColor="@color/black"/>
+                                android:textColor="@color/black" />
 
                             <TextView
-                                android:layout_height="wrap_content"
                                 android:layout_width="match_parent"
+                                android:layout_height="wrap_content"
                                 android:text="@string/view_qr_code"
-                                android:textAppearance="?android:textAppearanceSmall"/>
+                                android:textAppearance="?android:textAppearanceSmall" />
```
Comment (luckyman20) [inline app/src/main/res/layout/fragment_saving_account_details.xml]: Why this space?

File: app/src/main/res/layout/fragment_saving_account_details.xml [hunk 3]
```diff
@@ -184,220 +181,220 @@
...
                             android:orientation="vertical">
 
                             <TextView
-                                android:layout_height="wrap_content"
                                 android:layout_width="match_parent"
+                                android:layout_height="wrap_content"
                                 android:text="@string/qr_code"
                                 android:textAppearance="?android:textAppearanceMedium"
-                                android:textColor="@color/black"/>
+                                android:textColor="@color/black" />
```
Comment (luckyman20) [inline app/src/main/res/layout/fragment_saving_account_details.xml]: Spacing?

File: app/src/main/res/layout/fragment_saving_account_details.xml [hunk 4]
```diff
@@ -184,220 +181,220 @@
...
 
                             <TextView
-                                android:layout_height="wrap_content"
                                 android:layout_width="match_parent"
+                                android:layout_height="wrap_content"
                                 android:text="@string/view_charges"
-                                android:textAppearance="?android:textAppearanceSmall"/>
+                                android:textAppearance="?android:textAppearanceSmall" />
```
Comment (luckyman20) [inline app/src/main/res/layout/fragment_saving_account_details.xml]: Spacing?

File: app/src/main/res/layout/fragment_saving_account_details.xml [hunk 5]
```diff
@@ -184,220 +181,220 @@
...
                             android:orientation="vertical">
 
                             <TextView
-                                android:layout_height="wrap_content"
                                 android:layout_width="match_parent"
+                                android:layout_height="wrap_content"
                                 android:text="@string/savings_charges"
                                 android:textAppearance="?android:textAppearanceMedium"
-                                android:textColor="@color/black"/>
+                                android:textColor="@color/black" />
```
Comment (luckyman20) [inline app/src/main/res/layout/fragment_saving_account_details.xml]: Spacing?

File: app/src/main/res/layout/fragment_saving_account_details.xml [hunk 6]
```diff
@@ -184,220 +181,220 @@
...
 
                             <TextView
-                                android:layout_height="wrap_content"
                                 android:layout_width="match_parent"
+                                android:layout_height="wrap_content"
                                 android:text="@string/view_transactions"
-                                android:textAppearance="?android:textAppearanceSmall"/>
+                                android:textAppearance="?android:textAppearanceSmall" />
```
Comment (luckyman20) [inline app/src/main/res/layout/fragment_saving_account_details.xml]: Spacing?

File: app/src/main/res/layout/fragment_saving_account_details.xml [hunk 7]
```diff
@@ -184,220 +181,220 @@
...
                             android:orientation="vertical">
 
                             <TextView
-                                android:layout_height="wrap_content"
                                 android:layout_width="match_parent"
+                                android:layout_height="wrap_content"
                                 android:text="@string/transactions"
                                 android:textAppearance="?android:textAppearanceMedium"
-                                android:textColor="@color/black"/>
+                                android:textColor="@color/black" />
```
Comment (luckyman20) [inline app/src/main/res/layout/fragment_saving_account_details.xml]: Spacing?

File: app/src/main/res/layout/fragment_saving_account_details.xml [hunk 8]
```diff
@@ -184,220 +181,220 @@
...
                         android:id="@+id/made_on"
+                        android:layout_width="wrap_content"
                         android:layout_height="wrap_content"
                         android:layout_marginTop="@dimen/default_margin"
-                        android:layout_width="wrap_content"
+                        android:ellipsize="end"
                         android:text="@string/made_on"
                         android:textAppearance="?android:textAppearanceMedium"
-                        android:textColor="@color/black"/>
+                        android:textColor="@color/black" />
```
Comment (luckyman20) [inline app/src/main/res/layout/fragment_saving_account_details.xml]: This space also

File: app/src/main/res/layout/fragment_saving_account_details.xml [hunk 9]
```diff
@@ -184,220 +181,220 @@
...
-                        android:ellipsize="end"
                         android:id="@+id/last_trans"
-                        android:layout_height="wrap_content"
                         android:layout_width="match_parent"
+                        android:layout_height="wrap_content"
+                        android:ellipsize="end"
                         android:text="@string/last_transaction"
                         android:textAppearance="?android:textAppearanceMedium"
-                        android:textColor="@color/black"/>
+                        android:textColor="@color/black" />
```
Comment (luckyman20) [inline app/src/main/res/layout/fragment_saving_account_details.xml]: This space also

File: app/src/main/res/layout/fragment_saving_account_details.xml [hunk 10]
```diff
@@ -57,125 +57,122 @@
...
                             android:id="@+id/tv_make_a_transfer"
-                            android:layout_gravity="end"
-                            android:layout_height="wrap_content"
                             android:layout_width="wrap_content"
+                            android:layout_height="wrap_content"
+                            android:layout_gravity="end"
                             android:text="@string/make_transfer"
                             android:textAppearance="?android:textAppearanceMedium"
-                            android:textColor="@android:color/white"/>
+                            android:textColor="@android:color/white" />
```
Comment (luckyman20) [inline app/src/main/res/layout/fragment_saving_account_details.xml]: This space also

File: app/src/main/res/layout/fragment_saving_account_details.xml [hunk 11]
```diff
@@ -57,125 +57,122 @@
...
 
                         <android.support.v7.widget.AppCompatButton
                             android:id="@+id/tv_deposit"
-                            android:layout_height="wrap_content"
                             android:layout_width="wrap_content"
+                            android:layout_height="wrap_content"
                             android:text="@string/deposit"
                             android:textAppearance="?android:textAppearanceMedium"
-                            android:textColor="@android:color/white"/>
+                            android:textColor="@android:color/white" />
```
Comment (luckyman20) [inline app/src/main/res/layout/fragment_saving_account_details.xml]: This space also

File: app/src/main/res/layout/fragment_saving_account_details.xml [hunk 12]
```diff
@@ -57,125 +57,122 @@
...
                         android:layout_height="wrap_content"
                         android:layout_marginTop="@dimen/default_margin"
                         android:layout_weight="1"
-                        android:layout_width="wrap_content"
+                        android:ellipsize="end"
                         android:lines="1"
                         android:text="@string/total_withdrawal"
                         android:textAppearance="?android:textAppearanceMedium"
-                        android:textColor="@color/black"/>
+                        android:textColor="@color/black" />
```
Comment (luckyman20) [inline app/src/main/res/layout/fragment_saving_account_details.xml]: This space also

File: app/src/main/res/layout/fragment_saving_account_details.xml [hunk 13]
```diff
@@ -57,125 +57,122 @@
...
                         android:layout_height="wrap_content"
                         android:layout_marginTop="@dimen/default_margin"
                         android:layout_weight="1"
-                        android:layout_width="wrap_content"
+                        android:ellipsize="end"
                         android:lines="1"
                         android:text="@string/total_deposits"
                         android:textAppearance="?android:textAppearanceMedium"
-                        android:textColor="@color/black"/>
+                        android:textColor="@color/black" />
```
Comment (luckyman20) [inline app/src/main/res/layout/fragment_saving_account_details.xml]: This space also

File: app/src/main/res/layout/fragment_saving_account_details.xml [hunk 14]
```diff
@@ -57,125 +57,122 @@
...
                         android:layout_height="wrap_content"
                         android:layout_marginTop="@dimen/default_margin"
                         android:layout_weight="1"
-                        android:layout_width="wrap_content"
+                        android:ellipsize="end"
                         android:lines="1"
                         android:text="@string/nominal_interest_rate"
                         android:textAppearance="?android:textAppearanceMedium"
-                        android:textColor="@color/black"/>
+                        android:textColor="@color/black" />
```
Comment (luckyman20) [inline app/src/main/res/layout/fragment_saving_account_details.xml]: This space also

File: app/src/main/res/layout/fragment_saving_account_details.xml [hunk 15]
```diff
@@ -57,125 +57,122 @@
...
                         android:layout_height="wrap_content"
                         android:layout_marginTop="@dimen/default_margin"
                         android:layout_weight="1"
-                        android:layout_width="wrap_content"
+                        android:ellipsize="end"
                         android:lines="1"
                         android:text="@string/account_number"
                         android:textAppearance="?android:textAppearanceMedium"
-                        android:textColor="@color/black"/>
+                        android:textColor="@color/black" />
```
Comment (luckyman20) [inline app/src/main/res/layout/fragment_saving_account_details.xml]: This space also

File: app/src/main/res/layout/fragment_saving_account_details.xml [hunk 16]
```diff
@@ -57,125 +57,122 @@
...
+                            android:visibility="visible" />
 
                         <TextView
                             android:id="@+id/tv_account_status"
-                            android:layout_height="wrap_content"
                             android:layout_width="wrap_content"
+                            android:layout_height="wrap_content"
                             android:textAppearance="?android:textAppearanceSmall"
-                            android:visibility="visible"/>
+                            android:visibility="visible" />
```
Comment (luckyman20) [inline app/src/main/res/layout/fragment_saving_account_details.xml]: This space also

File: app/src/main/res/layout/fragment_saving_account_details.xml [hunk 17]
```diff
@@ -12,39 +13,38 @@
...
                         android:id="@+id/acc_balance"
+                        android:layout_width="match_parent"
                         android:layout_height="wrap_content"
                         android:layout_marginTop="@dimen/default_margin"
-                        android:layout_width="match_parent"
+                        android:ellipsize="end"
                         android:text="@string/account_balance"
                         android:textAppearance="?android:textAppearanceMedium"
-                        android:textColor="@color/black"/>
+                        android:textColor="@color/black" />
```
Comment (luckyman20) [inline app/src/main/res/layout/fragment_saving_account_details.xml]: This space also

**Review by luckyman20 — CHANGES_REQUESTED** [review]
@sashank27 Please remove the unnecessary spacing

File: app/src/main/res/layout/fragment_saving_account_details.xml (hunk 1 above)
Comment (sashank27) [reply app/src/main/res/layout/fragment_saving_account_details.xml]: @luckyman20 these all spacing came after I did ./gradlew check and ./gradlew build. Will edit these and send new PR.

Comment (luckyman20) [reply app/src/main/res/layout/fragment_saving_account_details.xml]: @sashank27 No need for making a new PR, just make the changes and commit them

Comment (sashank27) [reply app/src/main/res/layout/fragment_saving_account_details.xml]: ok

**Comment by sashank27** [comment]
@dilpreet96 @luckyman20 please review

File: app/src/main/res/layout/layout_error.xml [hunk 18]
```diff
@@ -20,7 +20,7 @@
         android:layout_height="wrap_content"
         android:layout_below="@+id/iv_status"
         android:layout_centerHorizontal="true"
-        android:layout_centerVertical="true"
+        android:padding="10dp"
```
Comment (luckyman20) [inline app/src/main/res/layout/layout_error.xml:23]: @sashank27 Why you are using padding here instead of `android:layout_centerVertical="true"`

Comment (sashank27) [reply app/src/main/res/layout/layout_error.xml:23]: Actually, `android:layout_centerVertical="true"` causes the overlapping. So it has to be removed. And I added the padding so as to make them distinct

File: app/src/main/res/layout/fragment_saving_account_details.xml [hunk 19]
```diff
@@ -2,8 +2,9 @@
 <android.support.v4.widget.NestedScrollView
     xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
-    android:layout_height="wrap_content"
-    android:layout_width="match_parent">
+    android:layout_height="match_parent"
```
Comment (luckyman20) [inline app/src/main/res/layout/fragment_saving_account_details.xml:5]: @sashank27 Why you made the wrap_content to match_parent?

Comment (sashank27) [reply app/src/main/res/layout/fragment_saving_account_details.xml:5]: @luckyman20 making it match_parent and doing `android:fillViewport="true"` enables the online layout to implement correct scrolling behaviour, and also makes the offline layout centred.

**Comment by therajanmaurya** [comment]
We are using https://github.com/therajanmaurya/Sweet-Error for handling such things please implement using sweet error and remove your error showing implemented UI.

**Comment by therajanmaurya** [comment]
Already fixed.

**Pull request closed without merging**